onnxruntime/core/graph/contrib_ops/nhwc_schema_defs.cc

```diff
@@ -383,5 +383,41 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
     NhwcConv,
     1,
     OpSchema().FillUsing(ConvOpSchemaGenerator()));
+
+ONNX_MS_OPERATOR_SET_SCHEMA(NhwcFusedConv, 1,
+                            OpSchema()
+                                .SetDoc(R"DOC(
+NhwcFusedConv is a float16 Conv operator with optional activation and add operators fused in.
+It also processes both NHWC and NCHW format.
+)DOC")
+                                .Attr("channels_last", "0(default) when input tensor is NCHW, else the input is NHWC", AttributeProto::INT, static_cast<int64_t>(0))
+                                .Attr("auto_pad", "", AttributeProto::STRING, std::string("NOTSET"))
```

Comment:
The name already has Nhwc, which indicates the op is channel last.

Reply:
Which activations are you wanting to support? The registration for the NHWC Conv in the internal NHWC domain has both activation and activation_params. 

[https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/graph/contrib_ops/internal_nhwc_onnx_opset.cc#L90](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/graph/contrib_ops/internal_nhwc_onnx_opset.cc#L90)

[https://github.com/microsoft/onnxruntime/blob/198994d01d4e226fd3a8816e8b2fad00c84a691c/onnxruntime/core/graph/contrib_ops/internal_nhwc_onnx_opset.cc#L42-L43](https://github.com/microsoft/onnxruntime/blob/198994d01d4e226fd3a8816e8b2fad00c84a691c/onnxruntime/core/graph/contrib_ops/internal_nhwc_onnx_opset.cc#L42-L43)

So if you only need to add an NHWC format Conv that can fuse with an activation, the existing registration gives you one based on the ONNX schema (which includes float16) with ONNX type/shape inferencing.

Reply:
Two things. 1, FusedConv have not only activation, but also an Add operator fused in.  2. I don't fully understand the rationale of this part to be honest. In my mind the schema is a contract. And our implementations of the operators must adhere to the contracts. It seems to me only the xnnpack EP's internal nhwc conv operator support activation attributes? What about other implementations?